daffodil-runtime1/src/main/scala/org/apache/daffodil/infoset/InfosetImpl.scala

```diff
@@ -1409,6 +1431,18 @@ sealed class DIComplex(override val erd: ElementRuntimeData, val tunable: Daffod
     e.setParent(this)
   }
 
+  def maybeMostRecentlyAddedChild(): Maybe[DIElement] = {
```

Comment:
This is kind of a hack. Suggestions welcome for what to do instead. At the point we need the child element just added, we've already popped that node from the currentElement stack and the focus is back on the parent. I did not want to change the invariant about which element is current.